### ISSUE TITLE
1.3rc1: various issues with astropy.io.misc.yaml on RHEL & py2.7

### ISSUE THREAD

**Issue by bsipocz** [astropy/astropy#5602]
I'm getting the following 6 issues when testing on a relatively old server (but as I see all the dependencies are still within the supported version range).

```
============================= test session starts ==============================
platform linux2 -- Python 2.7.8, pytest-2.8.3, py-1.4.30, pluggy-0.3.1

Running tests with Astropy version 1.3rc1.
Running tests in astropy.

Date: 2016-12-13T11:58:48

Platform: Linux-2.6.18-416.el5-x86_64-with-redhat-5.11-Tikanga

Executable: /data/apm36_a/wts/pipeline64/bsipocz/bin/python

Full Python Version: 
2.7.8 (default, Nov 18 2014, 11:18:05) 
[GCC 4.1.2 20080704 (Red Hat 4.1.2-55)]

encodings: sys: ascii, locale: UTF-8, filesystem: UTF-8, unicode bits: 15
byteorder: little
float info: dig: 15, mant_dig: 15

Numpy: 1.11.0
Scipy: 0.16.0
Matplotlib: 1.4.2
h5py: not available
Pandas: 0.17.0
Cython: 0.23.2
Using Astropy options: remote_data.

rootdir: /home/bsipocz/devel/astropy, inifile: setup.cfg
collected 11821 items 

...
=================================== FAILURES ===================================
_________________________ test_ndarray_subclasses[c4] __________________________

c = array([[1, 2],
       [3, 4]])

    @pytest.mark.parametrize('c', [Angle('1 2 3', unit='deg'),
                                   Longitude('1 2 3', unit='deg'),
                                   Latitude('1 2 3', unit='deg'),
                                   [[1], [3]] * u.m,
                                   np.array([[1, 2], [3, 4]], order='F'),
                                   np.array([[1, 2], [3, 4]], order='C'),
                                   np.array([1, 2, 3, 4])[::2]])
    def test_ndarray_subclasses(c):
>       cy = load(dump(c))

astropy/io/misc/tests/test_yaml.py:42: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/io/misc/yaml.py:285: in dump
    return yaml.dump(data, stream=stream, **kwargs)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/__init__.py:202: in dump
    return dump_all([data], stream, Dumper=Dumper, **kwds)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/__init__.py:190: in dump_all
    dumper.represent(data)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:28: in represent
    node = self.represent_data(data)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:41: in represent_data
    if self.ignore_aliases(data):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <astropy.io.misc.yaml.AstropyDumper object at 0x2b99011fbf10>
data = array([[1, 2],
       [3, 4]])

    def ignore_aliases(self, data):
>       if data in [None, ()]:
E       DeprecationWarning: elementwise == comparison failed; this will raise an error in the future.

/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:142: DeprecationWarning
_________________________ test_ndarray_subclasses[c5] __________________________

c = array([[1, 2],
       [3, 4]])

    @pytest.mark.parametrize('c', [Angle('1 2 3', unit='deg'),
                                   Longitude('1 2 3', unit='deg'),
                                   Latitude('1 2 3', unit='deg'),
                                   [[1], [3]] * u.m,
                                   np.array([[1, 2], [3, 4]], order='F'),
                                   np.array([[1, 2], [3, 4]], order='C'),
                                   np.array([1, 2, 3, 4])[::2]])
    def test_ndarray_subclasses(c):
>       cy = load(dump(c))

astropy/io/misc/tests/test_yaml.py:42: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/io/misc/yaml.py:285: in dump
    return yaml.dump(data, stream=stream, **kwargs)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/__init__.py:202: in dump
    return dump_all([data], stream, Dumper=Dumper, **kwds)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/__init__.py:190: in dump_all
    dumper.represent(data)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:28: in represent
    node = self.represent_data(data)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:41: in represent_data
    if self.ignore_aliases(data):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <astropy.io.misc.yaml.AstropyDumper object at 0x2b9900ce77d0>
data = array([[1, 2],
       [3, 4]])

    def ignore_aliases(self, data):
>       if data in [None, ()]:
E       DeprecationWarning: elementwise == comparison failed; this will raise an error in the future.

/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:142: DeprecationWarning
_________________________ test_ndarray_subclasses[c6] __________________________

c = array([1, 3])

    @pytest.mark.parametrize('c', [Angle('1 2 3', unit='deg'),
                                   Longitude('1 2 3', unit='deg'),
                                   Latitude('1 2 3', unit='deg'),
                                   [[1], [3]] * u.m,
                                   np.array([[1, 2], [3, 4]], order='F'),
                                   np.array([[1, 2], [3, 4]], order='C'),
                                   np.array([1, 2, 3, 4])[::2]])
    def test_ndarray_subclasses(c):
>       cy = load(dump(c))

astropy/io/misc/tests/test_yaml.py:42: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/io/misc/yaml.py:285: in dump
    return yaml.dump(data, stream=stream, **kwargs)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/__init__.py:202: in dump
    return dump_all([data], stream, Dumper=Dumper, **kwds)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/__init__.py:190: in dump_all
    dumper.represent(data)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:28: in represent
    node = self.represent_data(data)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:41: in represent_data
    if self.ignore_aliases(data):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <astropy.io.misc.yaml.AstropyDumper object at 0x2b99011fa9d0>
data = array([1, 3])

    def ignore_aliases(self, data):
>       if data in [None, ()]:
E       DeprecationWarning: elementwise == comparison failed; this will raise an error in the future.

/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:142: DeprecationWarning
______________________________ test_skycoord[fk4] ______________________________

frame = 'fk4'

    @pytest.mark.parametrize('frame', ['fk4', 'altaz'])
    def test_skycoord(frame):
    
        c = SkyCoord([[1,2],[3,4]], [[5,6], [7,8]],
                     unit='deg', frame=frame,
                     obstime=Time('2016-01-02'),
                     location=EarthLocation(1000, 2000, 3000, unit=u.km))
>       cy = load(dump(c))

astropy/io/misc/tests/test_yaml.py:83: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/io/misc/yaml.py:285: in dump
    return yaml.dump(data, stream=stream, **kwargs)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/__init__.py:202: in dump
    return dump_all([data], stream, Dumper=Dumper, **kwds)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/__init__.py:190: in dump_all
    dumper.represent(data)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:28: in represent
    node = self.represent_data(data)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:57: in represent_data
    node = self.yaml_representers[data_types[0]](self, data)
astropy/io/misc/yaml.py:161: in _skycoord_representer
    map)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:123: in represent_mapping
    node_value = self.represent_data(item_value)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:41: in represent_data
    if self.ignore_aliases(data):
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:142: in ignore_aliases
    if data in [None, ()]:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Latitude [[ 5., 6.],
           [ 7., 8.]] deg>, other = ()

    def __eq__(self, other):
        try:
            try:
                return super(Quantity, self).__eq__(other)
            except DeprecationWarning:
                # We treat the DeprecationWarning separately, since it may
                # mask another Exception.  But we do not want to just use
                # np.equal, since super's __eq__ treats recarrays correctly.
>               return np.equal(self, other)
E               ValueError: operands could not be broadcast together with shapes (2,2) (0,)

astropy/units/quantity.py:897: ValueError
_____________________________ test_skycoord[altaz] _____________________________

frame = 'altaz'

    @pytest.mark.parametrize('frame', ['fk4', 'altaz'])
    def test_skycoord(frame):
    
        c = SkyCoord([[1,2],[3,4]], [[5,6], [7,8]],
                     unit='deg', frame=frame,
                     obstime=Time('2016-01-02'),
                     location=EarthLocation(1000, 2000, 3000, unit=u.km))
>       cy = load(dump(c))

astropy/io/misc/tests/test_yaml.py:83: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/io/misc/yaml.py:285: in dump
    return yaml.dump(data, stream=stream, **kwargs)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/__init__.py:202: in dump
    return dump_all([data], stream, Dumper=Dumper, **kwds)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/__init__.py:190: in dump_all
    dumper.represent(data)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:28: in represent
    node = self.represent_data(data)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:57: in represent_data
    node = self.yaml_representers[data_types[0]](self, data)
astropy/io/misc/yaml.py:161: in _skycoord_representer
    map)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:123: in represent_mapping
    node_value = self.represent_data(item_value)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:41: in represent_data
    if self.ignore_aliases(data):
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:142: in ignore_aliases
    if data in [None, ()]:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Latitude [[ 5., 6.],
           [ 7., 8.]] deg>, other = ()

    def __eq__(self, other):
        try:
            try:
                return super(Quantity, self).__eq__(other)
            except DeprecationWarning:
                # We treat the DeprecationWarning separately, since it may
                # mask another Exception.  But we do not want to just use
                # np.equal, since super's __eq__ treats recarrays correctly.
>               return np.equal(self, other)
E               ValueError: operands could not be broadcast together with shapes (2,2) (0,)

astropy/units/quantity.py:897: ValueError
________________________________ test_load_all _________________________________

    def test_load_all():
        t = _get_time()
        unit = u.m / u.s
        c = SkyCoord([[1,2],[3,4]], [[5,6], [7,8]],
                     unit='deg', frame='fk4',
                     obstime=Time('2016-01-02'),
                     location=EarthLocation(1000, 2000, 3000, unit=u.km))
    
        # Make a multi-document stream
        out = ('---\n' + dump(t)
               + '---\n' + dump(unit)
>              + '---\n' + dump(c))

astropy/io/misc/tests/test_yaml.py:134: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/io/misc/yaml.py:285: in dump
    return yaml.dump(data, stream=stream, **kwargs)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/__init__.py:202: in dump
    return dump_all([data], stream, Dumper=Dumper, **kwds)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/__init__.py:190: in dump_all
    dumper.represent(data)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:28: in represent
    node = self.represent_data(data)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:57: in represent_data
    node = self.yaml_representers[data_types[0]](self, data)
astropy/io/misc/yaml.py:161: in _skycoord_representer
    map)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:123: in represent_mapping
    node_value = self.represent_data(item_value)
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:41: in represent_data
    if self.ignore_aliases(data):
/data/apm36_a/wts/pipeline64/bsipocz/lib/python2.7/site-packages/yaml/representer.py:142: in ignore_aliases
    if data in [None, ()]:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Latitude [[ 5., 6.],
           [ 7., 8.]] deg>, other = ()

    def __eq__(self, other):
        try:
            try:
                return super(Quantity, self).__eq__(other)
            except DeprecationWarning:
                # We treat the DeprecationWarning separately, since it may
                # mask another Exception.  But we do not want to just use
                # np.equal, since super's __eq__ treats recarrays correctly.
>               return np.equal(self, other)
E               ValueError: operands could not be broadcast together with shapes (2,2) (0,)

astropy/units/quantity.py:897: ValueError
 6 failed, 11628 passed, 141 skipped, 43 xfailed, 3 xpassed, 2 pytest-warnings in 720.74 seconds 
```

**Comment by mhvk** [comment]
@bsipocz - I think #5603 should fix it, but please have a look

**Comment by mhvk** [comment]
fixed by #5603, so closing
